src/plugins/intel_gpu/src/graph/graph_optimizer/prepare_buffer_fusing.cpp

```diff
@@ -134,32 +137,33 @@ bool concat_in_place_optimization::match(concatenation_node& node) {
         if (output_format != l.format || output_datatype != l.data_type)
             return false;
 
-        // TODO: Below condition should be moved to program_node::supports_padding.
-        // This hovewer will need updating the algorithm as it may make cascade adjustment impossible in some cases.
-        // It hovewer would make normal optimizations possible in others, so this is a trade-off to be investigated.
-        if (l.format == format::b_fs_yx_fsv16 && (l.size.feature[0] % 16 != 0 || node.get_primitive()->axis != concatenation::along_f))
-            return false;
-
-        if (l.format == format::b_fs_zyx_fsv16 && (l.size.feature[0] % 16 != 0 || node.get_primitive()->axis != concatenation::along_f))
+        if (l.format == format::bs_fs_yx_bsv16_fsv16)
```

Comment:
why do you treat `bs_fs_yx_bsv16_fsv16` specially? what about other double blocked formats, such as `bsv32_fsv16` or `bsv32_fsv32`?
But more importantly, it would be better to check with white list, I think...

Reply:
Agree, I'll update to check blocks size and only available 1.